test/flows/student_finance_calculator_flow_test.rb

```diff
@@ -0,0 +1,362 @@
+        assert_rendered_question
+      end
+
+      context "next_node" do
+        should "have a next node of what_course_are_you_studying?" do
+          assert_next_node :what_course_are_you_studying?, for_response: "has-disability,low-income"
+        end
+      end
+    end
+
+    context "what_course_are_you_studying?" do
+      setup do
+        testing_node :what_course_are_you_studying?
+        add_responses when_does_your_course_start?: "2020-2021",
+                      what_type_of_student_are_you?: "uk-part-time",
+                      how_much_are_your_tuition_fees_per_year?: "6935",
+                      where_will_you_live_while_studying?: "at-home",
+                      whats_your_household_income?: "50,000",
+                      how_many_credits_will_you_study?: "10",
+                      how_many_credits_does_a_full_time_student_study?: "11",
+                      do_any_of_the_following_apply_all_uk_students?: "has-disability,low-income"
+      end
+
+      should "render the question" do
+        assert_rendered_question
+      end
+
+      context "next_node" do
+        should "have a next node of outcome_uk_part_time_students for any response" do
+          assert_next_node :outcome_uk_part_time_students, for_response: "dental-medical-healthcare"
+        end
+      end
+    end
+  end
+
+  context "EU full time students" do
+    context "question: how_much_are_your_tuition_fees_per_year?," do
+      setup do
+        testing_node :how_much_are_your_tuition_fees_per_year?
+        add_responses when_does_your_course_start?: "2020-2021",
+                      what_type_of_student_are_you?: "eu-full-time"
+      end
+
+      should "render the question" do
+        assert_rendered_question
+      end
+
+      context "validation" do
+        should "be invalid if a fee over 9250 is entered for full time students" do
+          assert_invalid_response "9251"
+        end
+      end
+
+      context "next_node" do
+        should "have a next node of outcome_eu_students" do
+          assert_next_node :outcome_eu_students, for_response: "9250"
+        end
+      end
+    end
+  end
+
+  context "EU part time students" do
+    context "question: how_much_are_your_tuition_fees_per_year?," do
+      setup do
+        testing_node :how_much_are_your_tuition_fees_per_year?
+        add_responses when_does_your_course_start?: "2020-2021",
+                      what_type_of_student_are_you?: "eu-part-time"
+      end
+
+      should "render the question" do
+        assert_rendered_question
+      end
+
+      context "validation" do
+        should "be invalid if a fee over 6935 is entered for part time students" do
+          assert_invalid_response "6936"
+        end
+      end
+
+      context "next_node" do
+        should "have a next node of outcome_eu_students" do
+          assert_next_node :outcome_eu_students, for_response: "6935"
+        end
+      end
+    end
+  end
+end
```

Comment:
We should add tests for the logic in the outcomes. Some examples of this are:

https://github.com/alphagov/smart-answers/blob/65c061b5fb62ba0d81267bd269b492d4789cdae0/app/flows/student_finance_calculator_flow/outcomes/outcome_uk_full_time_students.erb#L10-L48
https://github.com/alphagov/smart-answers/blob/65c061b5fb62ba0d81267bd269b492d4789cdae0/app/flows/student_finance_calculator_flow/outcomes/_nhs_full_time_funding.erb#L3-L29
https://github.com/alphagov/smart-answers/blob/65c061b5fb62ba0d81267bd269b492d4789cdae0/app/flows/student_finance_calculator_flow/outcomes/_tuition_maintenance_summary.erb#L5-L19

There's a brief guide on how to test these in the ADR: https://github.com/alphagov/smart-answers/blob/main/docs/arch/003-testing-flows.md#testing-logic-in-an-outcome. The motivation for doing this is to explore all the executable code in the test suite, which basically offers a guarantee against any basic errors (such as syntax).

But for a good example of it applied to an actual flow see https://github.com/alphagov/smart-answers/blob/65c061b5fb62ba0d81267bd269b492d4789cdae0/test/flows/uk_benefits_abroad_flow_test.rb#L1170-L1196